public/app/features/browse-dashboards/api/browseDashboardsAPI.ts

```diff
@@ -338,6 +338,11 @@ export const browseDashboardsAPI = createApi({
     saveDashboard: builder.mutation<SaveDashboardResponseDTO, SaveDashboardCommand>({
       queryFn: async (cmd) => {
         try {
+          // When we use the `useV2DashboardsAPI` flag, we can save 'v2' schema dashboards
+          if (config.featureToggles.useV2DashboardsAPI) {
+            const response = await getDashboardAPI('v2').saveDashboard(cmd);
```

Comment:
this seems like it's not going to work when we start using `getDashboardAPI` without version (basically version provided behind the scenes).

Reply:
`saveDashboard: builder.mutation<SaveDashboardResponseDTO, SaveDashboardCommand>({`

this is a type mismatch here. v2 `saveDashboard` cannot accept `SaveDashboardCommand` as is

Reply:
I push changes converting `SaveDashboardCommand` as a generic , had to modify couple of places but it seems to work